components/call_button/call_button.jsx

```diff
@@ -0,0 +1,134 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import PropTypes from 'prop-types';
+import React, {PureComponent} from 'react';
+import {injectIntl} from 'react-intl';
+
+import {intlShape} from 'utils/react_intl';
+
+import MenuWrapper from 'components/widgets/menu/menu_wrapper';
+import Menu from 'components/widgets/menu/menu';
+import CameraIcon from 'components/widgets/icons/camera_icon';
+
+const customStyles = {
+    left: 'inherit',
+    right: 0,
+    bottom: '100%',
+    top: 'auto',
+};
+
+class CallButton extends PureComponent {
+    static propTypes = {
+        currentChannel: PropTypes.object.isRequired,
+        channelMember: PropTypes.object,
+        intl: intlShape.isRequired,
+        locale: PropTypes.string.isRequired,
+        pluginCallMethods: PropTypes.arrayOf(PropTypes.object),
+    };
+
+    static defaultProps = {
+        pluginCallMethods: [],
+    };
+
+    constructor(props) {
+        super(props);
+        this.state = {
+            menuOpen: false,
+        };
+    }
+
+    toggleMenu = (open) => {
+        this.setState({menuOpen: open});
+    }
+
+    render() {
+        const {formatMessage} = this.props.intl;
+
+        let bodyAction;
+
+        if (this.props.pluginCallMethods.length === 0) {
+            bodyAction = null;
+        } else if (this.props.pluginCallMethods.length === 1) {
+            const item = this.props.pluginCallMethods[0];
+            bodyAction = (
+                <div>
+                    <button
+                        type='button'
+                        id='calldButton'
+                        className='style--none post-action icon icon--attachment'
+                        onClick={() => {
+                            if (item.action) {
+                                item.action(this.props.currentChannel, this.props.channelMember);
+                            }
+                        }}
+                        onTouchEnd={() => {
```

Comment:
Do we need a separate `onTouchEnd` for this? `onClick` events should still be fired for touch events as far as I know.

Reply:
This was copy pasted from the FileUpload component. Not sure about the reason it was implemented like this there. If you want, I can remove it.